.github/workflows/check_critic.yml

```diff
@@ -0,0 +1,16 @@
+---
+name: perlcritic
+on: [push, pull_request]
+
+jobs:
+  perlcritic_checks:
+    runs-on: ubuntu-latest
+    container:
+      image: registry.opensuse.org/devel/openqa/containers/os-autoinst_dev
+    steps:
+      - uses: actions/checkout@v4
+      - name: Static analysis
+        run: |
+          git config --global --add safe.directory '*'
+          zypper -n install perl-Perl-Critic
+          perlcritic $(git ls-files -- '*.p[ml]')
```

Comment:
i wonder whether i have to use https://github.com/os-autoinst/os-autoinst-common/blob/master/tools/perlcritic somehow here

Reply:
Maybe @josegomezr or @okurz can answer that

Reply:
`perlcritic` kinda "knows" what's a perl file, you can pass just the directories where it needs to be invoked.

In fact, the check is in the files you're pulling from `os-autoinst-common`. Use that as much as possible.

```yaml
# [...]
container:
      image: perldocker/perl-tester # this is now provided by os_autoinst dev container iirc
    steps:
      - uses: actions/checkout@v4
      - run: ./tools/perlcritic --quiet 
# [...]
```